src/hooks/useContributions.ts

```diff
@@ -69,36 +69,57 @@ const ensureSource =
     })),
   });
 
+const fetchContributions = async ({
+  url,
+  label,
+}: {
+  url: RequestInfo;
+  label: string;
+}) => {
+  let errorMessage;
+  try {
+    const response = await fetch(url);
+    if (!response.ok) {
+      throw new Error(`${response.status} ${response.statusText}`);
+    }
+    const jsonResponse = await response.json();
+    const modifiedResponse = ensureSource(WONDER)(jsonResponse);
+    return modifiedResponse;
+  } catch (err: any) {
+    errorMessage = err.message;
+  }
+
+  if (errorMessage) {
+    console.error(`Error fetching ${label} contributions:`, errorMessage, url);
+    throw new Error(errorMessage);
+  }
+};
+
 const deworkIntegration = (
   integration: Integration,
   timeInput: TimeInput
-): Promise<Response> => {
-  return fetch(
-    `https://api.deworkxyz.com/integrations/coordinape/${
-      integration.data.organizationId
-    }?epoch_start=${timeInput.startDate}&epoch_end=${
-      timeInput.endDate
-    }&workspace_ids=${encodeURIComponent(
-      integration.data.workspaceIds?.join(',') || ''
-    )}`
-  )
-    .then(res => res.json())
-    .then(ensureSource(DEWORK));
+): Promise<Response | undefined> => {
```

Comment:
Whats the intent of making the new`fetchContributions` func return undefined sometimes ? can't it just throw?

Reply:
otherwise this is good to go

Reply:
I made that way so that if the input to the hook is undefined, it will return undefined without trying to fetch from the API then we can handle the undefined data case in Front end.